lib/compile.js

```diff
@@ -105,6 +106,26 @@ function compileLoad(loader, load, compileOpts, cache) {
     if (format == 'cjs')
       mappedLoad.source = mappedLoad.source.replace(hashBangRegEx, '');
     return Promise.resolve()
+    .then(function () {
+      var metadata = mappedLoad.metadata;
+      if (metadata.sourceMap)
+        return;
+      var sourceMap = sourceMappingURL.getFrom(mappedLoad.source);
+      if (!sourceMap)
+        return;
+      mappedLoad.source = sourceMappingURL.removeFrom(mappedLoad.source);
+      if (sourceMap.startsWith('data:')) {
+        // inline source map
+        metadata.sourceMap = JSON.parse(dataUriToBuffer(sourceMap).toString('utf8'));
+      } else {
+        // relative path to the .map file
+        var sourcePath = path.resolve(fromFileURL(loader.baseURL), load.path);
+        var sourceMapPath = path.join(path.dirname(sourcePath), sourceMap);
+        return asp(fs.readFile)(sourceMapPath, 'utf8').then(function (sourceMap) {
+          metadata.sourceMap = JSON.parse(sourceMap);
```

Comment:
We allow `metadata.sourceMap` to be an object fine, so we should be able to skip the `JSON.parse` here.